packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -1493,11 +1493,11 @@ function testDomRouter(
           );
 
           expect(container.querySelector("form")?.getAttribute("action")).toBe(
-            "/foo/bar?a=1#hash"
+            "/foo?a=1#hash"
```

Comment:
This was the misunderstanding - splat portion should not be included here for now to match `Link` behavior